### ISSUE TITLE
machined thinks all machines terminate on "systemctl daemon-reload"

### ISSUE THREAD

**Issue by aburch** [systemd/systemd#376]
This applies at least to machines started via `systemd-nspawn` / `machinectl start`.

Executing `systemctl daemon-reload` results in `systemd-machined` believing the machine terminated, even though it's still running:

```
Jun 25 18:10:08 X systemd-machined[18606]: Machine X terminated.
```

I guess this happens due to the `UnitRemoved` signals emitted by systemd. `match_unit_removed` in `src/machine/machined-dbus.c` then releases the machine even though the unit comes back a bit later when a `UnitNew` signal is emitted.

I use systemd/220-7 from Debian's unstable suite. There's also a bug already filed in Debian: https://bugs.debian.org/788269

**Comment by aburch** [comment]
http://lists.freedesktop.org/archives/systemd-devel/2015-June/033357.html looks like the same issue and refers to 9b420b3cfb8b93daf50e4cdbc92b05f2209ef893 as the commit which might have introduced this as a regression.

**Comment by poettering** [comment]
There's a fix for this now in #898.
